tests/Functional/TestApplication/Controller/PersonController.php

```diff
@@ -122,6 +122,10 @@ public function personListForCountryAction(?Country $country = null)
     #[Route('/{id}/cars', 'person_cars_list')]
     #[AnnotationRoute('/{id}/cars', name: 'person_cars_list')]
     #[PurgeOn(Person::class, target: 'cars')]
+    #[PurgeOn(Person::class,
+        target: 'cars',
+        if: "obj.firstName === 'John'",
+    )]
     public function personCarsList(Person $person)
```

Comment:
@HypeMC i did't add separate test because we already have one specifically for nullable inverse subscriptions:
https://github.com/sofascore/purgatory-bundle/blob/a14ef9bf4e195fa9cd609a92de538af07dc55897/tests/Application/ApplicationTest.php#L630-L654

it just didn't fail because the PurgeOn attribute did not have `if`.
this test fails for added PurgeOn without changes from PR